src/account_manager/builder.rs

```diff
@@ -220,4 +220,13 @@ impl AccountManagerBuilder {
             storage_manager,
         })
     }
+
+    pub(crate) async fn from_account_manager(account_manager: &AccountManager) -> Self {
```

Comment:
Maybe just impl `From`? Bit more idiomatic

Reply:
Doesn't work since it needs to be async

Reply:
Ah right my bad overlooked that